### PR TITLE
Add (multithreaded) functions to the fingerprint generators for calculating multiple fingeprints in one call

### PR DESCRIPTION
**Pull request by greglandrum**
It's not clear that these would be immediately useful in C++, so I just put them in the Python wrapper code. We can pretty easily move them into the actual C++ library elsewhere if that seems right.

### CODE REVIEW

**Comment by bp-kelley** [comment]
We have some use cases where it would be nice to have these in C++, they would also be useful in the SWIG wrappers as well IMO.

**Comment by greglandrum** [comment]
> We have some use cases where it would be nice to have these in C++, they would also be useful in the SWIG wrappers as well IMO.

Ok; I'll move them

**Comment by greglandrum** [comment]
hmmm. `-Werror` is a gift that keeps on giving. <sigh>

File: Code/GraphMol/Fingerprints/FingerprintGenerator.cpp [hunk 1]
```diff
@@ -441,7 +447,112 @@ FingerprintGenerator<OutputType>::getFingerprint(
...
+  std::vector<std::uint32_t> *customBondInvariants = nullptr;
+  int confId = -1;
+  AdditionalOutput *additionalOutput = nullptr;
+  FingerprintFuncArguments args(fromAtoms, ignoreAtoms, confId,
+                                additionalOutput, customAtomInvariants,
+                                customBondInvariants);
+
+  std::vector<std::unique_ptr<ReturnType>> result;
+  auto numThreadsToUse = getNumThreadsToUse(numThreads);
+  unsigned int nmols = mols.size();
```
Comment (bp-kelley) [inline Code/GraphMol/Fingerprints/FingerprintGenerator.cpp:468]: could use result.reserve(mols);

**Review by bp-kelley — APPROVED** [review]
Two minor suggestions, otherwise LGTM.

**Comment by bp-kelley** [comment]
Nice!

**Pull request merged**